ballerina/xmldata.bal

```diff
@@ -172,8 +193,8 @@ isolated function getAttributesMap(json jTree, map<string> parentNamespaces = {}
     return attributes;
 }
 
-isolated function getNamespacesMap(json jTree, map<string> parentNamespaces = {}, JsonOptions options = {})
-                            returns map<string>|Error {
+isolated function getNamespacesMap(json jTree, map<string> allNamespaces, map<string> parentNamespaces = {},
```

Comment:
Rather passing the `allNamespaces` map. shall we return it.

Reply:
This function has to only return the namespaces which are related to that field. So it returns `namespaces `.

The `allNamespaces` is used to store all the namespaces. If the field name has a prefix, we have to get that URI of that attribute in the XML data generation. So, all values are stored in that to get value in https://github.com/ballerina-platform/module-ballerina-xmldata/pull/417/files#diff-ecda4f4acb566335c5b8a6598ba4ffc0002c3cf114c6867b20c16bb26d67780bR184.

Reply:
Fixed in [c09adf3](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/417/commits/c09adf3bcb9473defb6f0043c74ac72066d66072)